x-pack/plugins/siem/public/components/drag_and_drop/drag_drop_context_wrapper.tsx

```diff
@@ -37,78 +43,117 @@ interface Props {
+    addFieldToTimelineColumns({
+      browserFields,
+      dispatch,
+      result,
+      timelineId: ACTIVE_TIMELINE_REDUX_ID,
+    });
   }
 };
 
+const sensors = [useAddToTimelineSensor];
+
 /**
  * DragDropContextWrapperComponent handles all drag end events
  */
 export const DragDropContextWrapperComponent = React.memo<Props & PropsFromRedux>(
-  ({ browserFields, children, dataProviders, dispatch }) => {
+  ({ activeTimelineDataProviders, browserFields, children, dataProviders, dispatch }) => {
+    const [, dispatchToaster] = useStateToaster();
+    const onAddedToTimeline = useCallback(
+      (fieldOrValue: string) => {
+        displaySuccessToast(ADDED_TO_TIMELINE_MESSAGE(fieldOrValue), dispatchToaster);
+      },
+      [dispatchToaster]
+    );
+
     const onDragEnd = useCallback(
       (result: DropResult) => {
-        enableScrolling();
-
-        if (dataProviders != null) {
-          onDragEndHandler({
-            browserFields,
-            result,
-            dataProviders,
-            dispatch,
-          });
-        }
-
-        if (!draggableIsField(result)) {
+        try {
+          enableScrolling();
+
+          if (dataProviders != null) {
+            onDragEndHandler({
+              activeTimelineDataProviders,
+              browserFields,
+              dataProviders,
+              dispatch,
+              onAddedToTimeline,
+              result,
+            });
+          }
+        } finally {
           document.body.classList.remove(IS_DRAGGING_CLASS_NAME);
-        }
 
-        if (draggableIsField(result)) {
-          document.body.classList.remove(IS_TIMELINE_FIELD_DRAGGING_CLASS_NAME);
+          if (draggableIsField(result)) {
+            document.body.classList.remove(IS_TIMELINE_FIELD_DRAGGING_CLASS_NAME);
+          }
         }
       },
-      [browserFields, dataProviders]
+      [dataProviders, activeTimelineDataProviders, browserFields]
     );
     return (
       // @ts-ignore
-      <DragDropContext onDragEnd={onDragEnd} onBeforeCapture={onBeforeCapture}>
+      <DragDropContext onDragEnd={onDragEnd} onBeforeCapture={onBeforeCapture} sensors={sensors}>
         {children}
       </DragDropContext>
     );
   },
   (prevProps, nextProps) => {
     return (
       prevProps.children === nextProps.children &&
-      prevProps.dataProviders === nextProps.dataProviders
+      prevProps.dataProviders === nextProps.dataProviders &&
+      prevProps.activeTimelineDataProviders === nextProps.activeTimelineDataProviders
     ); // prevent re-renders when data providers are added or removed, but all other props are the same
   }
 );
 
 DragDropContextWrapperComponent.displayName = 'DragDropContextWrapperComponent';
 
 const emptyDataProviders: dragAndDropModel.IdToDataProvider = {}; // stable reference
+const emptyActiveTimelineDataProviders: DataProvider[] = []; // stable reference
 
 const mapStateToProps = (state: State) => {
+  const activeTimelineDataProviders =
```

Comment:
I know that you just add more features in this mapStateToProps but I am wondering if it will be a good idea to use `makMapStateToProps` to take advantage of the selectors. I think that can wait and fix it when we take care of the `browserfields` issue since everything is living in the same neighborhood.

Reply:
Agreed, I'll refactor this in the follow-up PR. 👍